src/vm/appdomain.hpp

```diff
@@ -1238,7 +1239,9 @@ class BaseDomain
     OBJECTHANDLE CreateTypedHandle(OBJECTREF object, int type)
     {
         WRAPPER_NO_CONTRACT;
-        return ::CreateTypedHandle(m_hHandleTableBucket->pTable[GetCurrentThreadHomeHeapNumber()], object, type);
+
+        IGCHandleTable *pHandleTable = GCHandleTableUtilities::GetGCHandleTable();
+        return pHandleTable->CreateHandleOfType(m_hHandleTableBucket->pTable[GetCurrentThreadHomeHeapNumber()], OBJECTREFToObject(object), type);
```

Comment:
HandleTableBucket seems to be implementation detail of the HandleTable. Shouldn't it move to the GC side as well?

Also, it does not look right that I have to call to GC two times to create the handle: first to get thread home number, and then to actually create the handle.